test/api/hh_sim.cpp

```diff
@@ -0,0 +1,97 @@
+// NOTE: this assumes neuronapi.h is on your CPLUS_INCLUDE_PATH
+#include "neuronapi.h"
+#include <dlfcn.h>
+
+#include <array>
+#include <cassert>
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+#include <fstream>
+
+#include "./test_common.h"
+
+using std::cout;
+using std::endl;
+using std::ofstream;
+
+constexpr std::array<double, 3> EXPECTED_V{
+#ifndef CORENEURON_ENABLED
+    -0x1.04p+6,
+    -0x1.b254ad82e20edp+5,
+    -0x1.24a52af1ab463p+6,
+#else
+    -0x1.04p+6,
+    -0x1.b0c75635b5bdbp+5,
+    -0x1.24a84bedb7246p+6,
+#endif
+};
+
+static const char* argv[] = {"hh_sim", "-nogui", "-nopython", nullptr};
+
+extern "C" void modl_reg(){};
+
+int main(void) {
+    Section* soma;
+    Object* iclamp;
+    Object* v;
+    Object* t;
+    char* temp_str;
+
+    nrn_init(3, argv);
```

Comment:
`argc` here should be `3` or `4` ? 🤔

Reply:
I might have made changes but kept the original `3`. Isn't it correct, given the `nullptr` is just this strange argv requirement the list must be terminated by null.